docs/tutorials/filters/AKKF.py

```diff
@@ -0,0 +1,381 @@
+# - :attr:`kernel`: the `Kernel` class which is chosen to be used to map the measurement space into
+#   the kernel space.
+
+# %%
+# Initialise a prior 2
+# --------------------
+#
+# To begin, we initialise a prior estimate for the tracking problem.
+# This is achieved by :class:`KernelParticleState` that describes the state as a distribution of
+# particles, each associated with StateVectors and corresponding weights.
+# The prior state is sampled from a Gaussian distribution.
+
+from scipy.stats import multivariate_normal
+
+from stonesoup.types.state import KernelParticleState
+from stonesoup.types.array import StateVectors
+
+number_particles = 30
+
+# Sample from the prior Gaussian distribution
+samples = multivariate_normal.rvs(np.squeeze(truth[0].state_vector),
+                                  np.diag([0.1, 0.005, 0.1, 0.01])**2,
+                                  size=number_particles)
+
+# Create prior particle state.
+prior = KernelParticleState(state_vector=StateVectors(samples.T),
+                            weight=np.array([1/number_particles]*number_particles),
+                            timestamp=start_time,
+                            )
+
+from stonesoup.kernel import QuadraticKernel
+from stonesoup.predictor.kernel import AdaptiveKernelKalmanPredictor
+predictor = AdaptiveKernelKalmanPredictor(transition_model=transition_model,
+                                          kernel=QuadraticKernel(c=1, ialpha=1))
+
+from stonesoup.updater.kernel import AdaptiveKernelKalmanUpdater
+updater = AdaptiveKernelKalmanUpdater(measurement_model=measurement_model,
+                                      kernel=QuadraticKernel(c=1, ialpha=1))
+
+# %%
+# Run the tracker
+# ---------------
+#
+# Now, we execute the predict and update steps of the tracker. The Predictor module takes both
+# the prior and proposal information to propagate the state particles, and it calculates the
+# predictive kernel weight vector and covariance matrix.
+#
+# Next, the Updater module uses the predictions to generate the measurement particles and updates
+# the posterior kernel weight vector and covariance matrix accordingly.
+# Additionally, the Updater generates new proposal particles at every step to refine the state
+# estimate.
+
+from stonesoup.types.hypothesis import SingleHypothesis
+from stonesoup.types.track import Track
+
+track = Track()
+for measurement in measurements:
+    prediction = predictor.predict(prior, timestamp=measurement.timestamp)
+    hypothesis = SingleHypothesis(prediction, measurement)
+    post = updater.update(hypothesis)
+    track.append(post)
+    prior = track[-1]
+
+# %%
+# Plot the resulting track with the sample points at each iteration.
+
+plotter = Plotter()
+plotter.plot_ground_truths(truth, [0, 2], linewidth=3.0, color='black')
+plotter.plot_tracks(track, [0, 2], track_label='AKKF - quadratic', color='royalblue')
+plotter.fig
+
+# %%
+
+plotter.plot_tracks(track, [0, 2], particle=True)
+plotter.fig
```

Comment:
Adding the comment below should use last image for gallery thumbnail, which I think is better.

```suggestion
# %%

# sphinx_gallery_thumbnail_number = -1

plotter.plot_tracks(track, [0, 2], track_label='AKKF - quadratic', color='royalblue', particle=True)
plotter.fig
```